### DISCUSSION TITLE
Dump your ideas here

### DISCUSSION THREAD

**Discussion by HakkaNgin** [usdevs/laundro#10] [General]
Some ideas that just struck me:

1. Are we able to configure the Raspberry Pi to be "dormant" till activated (i.e. not refreshing, sending signals etc. till it detects someone using the washing machine)? Could possibly save power and memory space
2. Since the washes are by default configured to 30 mins, how about we set a 30min timer after each "activation" of the sensor, and only update the system that the laundry is washed after the 30mins?

**Comment by shenyih0ng** [comment]
**1:** 

I believe we will not need to care too much about power since RPI does not really consume that much power anyways (https://www.pidramble.com/wiki/benchmarks/power-consumption) and also power is free since we don't pay for it 😅

But i do think it is definitely possible set up a "trigger" system but i would think that it will only come when we are confident that our system no longer have any false positives/false negatives since we will likely stop logging in its "dormant" state (or else it will defeat its purpose).

On the topic of memory space, we could also not keep sensor logs on the device itself and directly push it to a "central" logging system when network is stable. Then, in the case of network failure, we could store it on the client until it recovers, but i wonder if there is any value in that (unless we are trying to debug stuff)

**2:**

It is definitely the easiest way to make it work for the washers but the system will not be able to catch edges cases of people pausing the washers. I believe dryers can be configured to run 2 cycles so we might not be able to catch that easily with a fixed timer approach.

Another thing would also be the issue of sustainability, where perhaps in the future they might change the timing of the wash/allow double wash cycles then our system might either need manual tuning again or it will cease to be useful.

I guess we will want to look into the most generalizable (works for both washers and dryers) solution we can find that is sustainable as well.
